File: reinforcement.py

# Given an array of strings and an integer k, your task is to return the longest string consisting of 
# k consecutive strings from the array concatenated together


def longestConsecutive(arr, k):
    concactinate = []
    currentlongest = ""
    for i in range(len(arr)-k+1):
        concactinate.append(arr[i])
        for j in range(1,k):
            concactinate[i] = concactinate[i] + arr[i+j]
        if len(concactinate[i]) > len(currentlongest):
            currentlongest = concactinate[i]

    # print(concactinate)
    return(currentlongest)
        
print(longestConsecutive(['hi', 'marbles', 'mittens', 'bye', 'lorem', 'ipsum', 'to', 'a', 'hippocampus'], 3))
print(longestConsecutive(["zone", "abigail", "theta", "form", "libe", "zas", "theta", "abigail"], 2))

File: reinforcement.js
function longestConsecutive(arr, k){
    let con=[]
    let currentLongest = ""
    for(let i=0; i<=(arr.length-k); i++){
        con.push("")
        for(let j=0; j<k; j++){
            con[i] = con[i] + arr[i+j]
        }
        if(con[i].length > currentLongest.length){
            currentLongest = con[i]
        }
    }
    // console.log(con)
    // console.log(currentLongest)
    return(currentLongest)

}
console.log(longestConsecutive(['hi', 'marbles', 'mittens', 'bye', 'lorem', 'ipsum', 'to', 'a', 'hippocampus'], 3))
console.log(longestConsecutive(["zone", "abigail", "theta", "form", "libe", "zas", "theta", "abigail"], 2))